wasi/wasi.go

```diff
@@ -107,6 +110,48 @@ func Stderr(writer io.Writer) Option {
 	}
 }
 
+// wasiStringArray holds null-terminated strings. It ensures that
+// its length and total buffer size don't exceed the max of uint32.
+type wasiStringArray struct {
+	strings      [][]byte
+	totalBufSize uint32
+}
+
+// newWASIStringArray creates a cStringArray from the given string slice. It returns an error
```

Comment:
when using jargon like cstringarray add a `// See https://link` to describe what that helps with. The general assumption here will be that comparing to go things is something you can assume, so anything else, add a link!

Reply:
Nice catch! Actually, I just missed to fix `cStringArray` to `wasiStringArray`. `cStringArray` was an old name of this struct. Fixed it. Thanks!